pkg/segment/query/processor/searcher.go

```diff
@@ -63,6 +65,11 @@ type Searcher struct {
 	gotBlocks             bool
 	remainingBlocksSorted []*block // Sorted by time as specified by sortMode.
 	qsrs                  []*query.QuerySegmentRequest
+	segmentIdx            int
+	minTimestamp          uint64
+	processedBlocks       map[string]map[uint16]struct{}
+	processedSegment      map[string]struct{}
```

Comment:
nit: use the same pluralization for these two fields

Reply:
addressed in the latest commit.